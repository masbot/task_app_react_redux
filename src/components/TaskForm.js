import React, { Component } from 'react';
import { Form, FormGroup, FormControl, ControlLabel, Button } from 'react-bootstrap';

class TaskForm extends Component {
    constructor() {
        super();
        this.state = {
            task: ''
        }
    }

    render() {
        return (
            <div>
                <Form>
                    <FormGroup>
                        <FormControl 
                            placeholder="NEW TASK"
                            className="input-task" 
                            onChange={event => this.setState({task: event.target.value})}
                        />
                    </FormGroup>
                </Form>
                <Button className="btn btn-add-newtask" onClick={() => this.props.addNewTask(this.state.task)}>Submit</Button>
            </div>
        )
    }
}

export default TaskForm;