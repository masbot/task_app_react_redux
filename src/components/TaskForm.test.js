import React from 'react';
import { shallow } from 'enzyme';
import TaskForm from './TaskForm';

describe('TaskForm', () => {
    const task = 'BUY MILK';
    const mockAddNewTask = jest.fn();
    const props = { addNewTask: mockAddNewTask }
    const taskform = shallow(<TaskForm {...props}/>);

    it('renders correctly', () => {
        expect(taskform).toMatchSnapshot();
    });

    it('initalizes a task in `state`', () => {
        expect(taskform.state()).toEqual({task: ''});
    });

    describe('when typing into the task input', () => {
        beforeEach(() => {
            taskform.find('.input-task').simulate('change', { target: { value: task }});
        });

        it('updates the task in `state`', () => {
            expect(taskform.state().task).toEqual(task);
        });

        describe('and then the user wants to add the new task', () => {
            beforeEach(() => {
                taskform.find('.btn-add-newtask').simulate('click');
            });

            it('calls the addTask callback', () => {
                expect(mockAddNewTask).toHaveBeenCalledWith(task);
            });
        });
    });

});