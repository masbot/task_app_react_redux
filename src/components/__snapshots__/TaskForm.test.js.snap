// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskForm renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TaskForm
    addNewTask={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Form
          bsClass="form"
          componentClass="form"
          horizontal={false}
          inline={false}
>
          <FormGroup
                    bsClass="form-group"
          >
                    <FormControl
                              bsClass="form-control"
                              className="input-task"
                              componentClass="input"
                              onChange={[Function]}
                              placeholder="NEW TASK"
                    />
          </FormGroup>
</Form>,
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="btn btn-add-newtask"
          disabled={false}
          onClick={[Function]}
>
          Submit
</Button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "form",
          "children": <FormGroup
            bsClass="form-group"
>
            <FormControl
                        bsClass="form-control"
                        className="input-task"
                        componentClass="input"
                        onChange={[Function]}
                        placeholder="NEW TASK"
            />
</FormGroup>,
          "componentClass": "form",
          "horizontal": false,
          "inline": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-group",
            "children": <FormControl
              bsClass="form-control"
              className="input-task"
              componentClass="input"
              onChange={[Function]}
              placeholder="NEW TASK"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-control",
              "className": "input-task",
              "componentClass": "input",
              "onChange": [Function],
              "placeholder": "NEW TASK",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "block": false,
          "bsClass": "btn",
          "bsStyle": "default",
          "children": "Submit",
          "className": "btn btn-add-newtask",
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Submit",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Form
            bsClass="form"
            componentClass="form"
            horizontal={false}
            inline={false}
>
            <FormGroup
                        bsClass="form-group"
            >
                        <FormControl
                                    bsClass="form-control"
                                    className="input-task"
                                    componentClass="input"
                                    onChange={[Function]}
                                    placeholder="NEW TASK"
                        />
            </FormGroup>
</Form>,
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            className="btn btn-add-newtask"
            disabled={false}
            onClick={[Function]}
>
            Submit
</Button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form",
            "children": <FormGroup
              bsClass="form-group"
>
              <FormControl
                            bsClass="form-control"
                            className="input-task"
                            componentClass="input"
                            onChange={[Function]}
                            placeholder="NEW TASK"
              />
</FormGroup>,
            "componentClass": "form",
            "horizontal": false,
            "inline": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "children": <FormControl
                bsClass="form-control"
                className="input-task"
                componentClass="input"
                onChange={[Function]}
                placeholder="NEW TASK"
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-control",
                "className": "input-task",
                "componentClass": "input",
                "onChange": [Function],
                "placeholder": "NEW TASK",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": false,
            "bsClass": "btn",
            "bsStyle": "default",
            "children": "Submit",
            "className": "btn btn-add-newtask",
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Submit",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
